### PR TITLE
Roe 2392 bo delete warning params

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2392

### Change description

- config
- routes
- navigation
- controller
- unit tests

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/navigation.ts
Comment (dlloyd1-ch) [inline src/utils/navigation.ts:195]: I think this needs to be BENEFICIAL_OWNER_STATEMENTS_WITH_PARAMS_URL. Worth making sure you have the latest code from the feature branch and running it in Tilt.

File: test/controllers/beneficial.owner.delete.warning.controller.spec.ts
Comment (mwestacott) [inline test/controllers/beneficial.owner.delete.warning.controller.spec.ts:250]: Wondering if you should be making the same assertions as you made on lines 274 and 277 below, which would probably mean needing to do the mock call you did on line 263 too.

File: test/controllers/beneficial.owner.delete.warning.controller.spec.ts
Comment (mwestacott) [inline test/controllers/beneficial.owner.delete.warning.controller.spec.ts:277]: I think you should assert this too:

`expect(mockGetUrlWithParamsToPath).toHaveBeenCalledTimes(1);`

As we have done elsewhere.

Comment (mattch1) [reply test/controllers/beneficial.owner.delete.warning.controller.spec.ts:278]: Added this assertion.

File: test/controllers/beneficial.owner.delete.warning.controller.spec.ts
Comment (mattch1) [reply test/controllers/beneficial.owner.delete.warning.controller.spec.ts:250]: No, the flag condition is within the scope of the condition for 0 or 1.

File: src/utils/navigation.ts
Comment (dlloyd1-ch) [reply src/utils/navigation.ts:206]: ok, we can look at its usage in another task.

**Review by mwestacott — APPROVED** [review]
Happy with this, following clarifications. Suggest getting approval from @dlloyd1-ch too before merging, as he raised this sub-task and maybe also has another related change?

**Review by dlloyd1-ch — APPROVED** [review]
Looks good to me. My PR can follow, rather than be merged into this one.

**Pull request merged**
